### PR TITLE
DSS Operations: Checkout

### PR DESCRIPTION
**Pull request by amarjandu**
Adds dss-ops checkout: `start`, `verify`, `remove`
Fixes Testing Errors for `test/test_operations.py/Test_Checkout`

### CODE REVIEW

File: tests/test_operations.py [hunk 1]
```diff
@@ -418,5 +424,68 @@ def paginate(self):
...
+        cls.app = ThreadedLocalServer()
+        cls.app.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.app.shutdown()
+
+    def setUp(self):
+        Config.set_config(BucketConfig.TEST)
+        self.s3_test_bucket = get_env("DSS_S3_BUCKET_TEST")
```
Comment (amarjandu) [inline tests/test_operations.py]: Fix was here @DailyDreaming

File: dss/operations/checkout.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+            if DSS_BUNDLE_KEY_REGEX.match(_key):
+                uuid, version = self._parse_key(_key)
+                bundle_manifest = get_bundle_manifest(uuid=uuid, replica=self.replica, version=version)
+                self._sleepy_checkout(bundle_checkout, replica=self.replica, bundle_uuid=uuid, bundle_version=version)
+                for _files in bundle_manifest['files']:
+                    blob_path = compose_blob_key(_files)
+                    self._sleepy_checkout(file_checkout, replica=self.replica,
+                                          file_metadata=_files, blob_path=blob_path)
+            elif FILE_PREFIX in _key:
+                uuid, version = self._parse_key(_key)
```
Comment (DailyDreaming) [inline dss/operations/checkout.py]: This line is unused.

File: dss/operations/checkout.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        for _key in self.keys:
+            if DSS_BUNDLE_KEY_REGEX.match(_key):
+                uuid, version = self._parse_key(_key)
+                bundle_manifest = get_bundle_manifest(uuid=uuid, replica=self.replica, version=version)
+                self._sleepy_checkout(bundle_checkout, replica=self.replica, bundle_uuid=uuid, bundle_version=version)
+                for _files in bundle_manifest['files']:
+                    blob_path = compose_blob_key(_files)
+                    self._sleepy_checkout(file_checkout, replica=self.replica,
+                                          file_metadata=_files, blob_path=blob_path)
+            elif FILE_PREFIX in _key:
```
Comment (DailyDreaming) [inline dss/operations/checkout.py]: Should be `elif _key.startswith(FILE_PREFIX)`.  Odds are low that `"files"` might be a part of the `uuid`, but it's not impossible.

File: dss/operations/checkout.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+                    if bundle_key in checkout_bundle_contents:
+                        temp['bundle_checkout'] = True
+                    if cache_flow.should_cache_file(_file['content-type'], _file['size']):
+                        temp['should_be_cached'] = True
+
+                    for x in ['name', 'uuid', 'version']:
+                        temp.update({x: _file[x]})
+                    bundle_internal_status.append(temp)
+                checkout_status[_key] = bundle_internal_status
+            elif FILE_PREFIX in _key:
```
Comment (DailyDreaming) [inline dss/operations/checkout.py]: Should be elif _key.startswith(FILE_PREFIX). Odds are low that "files" might be a part of the uuid, but it's not impossible.

File: dss/operations/checkout.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+                    self._verify_delete(self.handle, self.bucket, key)
+                uuid, version = self._parse_key(_key)
+                manifest = get_bundle_manifest(replica=self.replica, uuid=uuid, version=version)
+                if manifest is None:
+                    logger.warning(f"unable to locate manifest for fqid: {self.bucket}/{_key}")
+                    continue
+                for _files in manifest['files']:
+                    key = compose_blob_key(_files)
+                    self._verify_delete(self.handle, self.bucket, key)
+            elif FILE_PREFIX in _key:
```
Comment (DailyDreaming) [inline dss/operations/checkout.py]: Should be elif _key.startswith(FILE_PREFIX). Odds are low that "files" might be a part of the uuid, but it's not impossible.

File: dss/operations/checkout.py (hunk 3 above)
Comment (amarjandu) [reply dss/operations/checkout.py]: done

**Pull request merged**